### DISCUSSION TITLE
Duration from the start of the trace to the end of a span (trace.duration_ms)

### DISCUSSION THREAD

**Discussion by srinusanduri** [open-telemetry/opentelemetry-java-instrumentation#10171] [Q&A]
We would like to compute the duration until end time of a span from the start of the trace (say `trace.duration_ms`). The `duration_ms` attribiute gives the duration of a span. Likewise, is there any attribute (or way) that will give us the `trace.duration_ms`, the duration until end time of a span from the start of the trace.

Here is an example to show the relation between `duration_ms` and `trace.duration_ms`.

<img width="699" alt="trace_duration_ms" src="https://github.com/open-telemetry/opentelemetry-java-instrumentation/assets/10183113/622efcc1-0aa4-46f1-a9b1-f1a31416adb0">

**Comment by trask** [comment]
typically this would be done on the backend (or maybe in a tail-based sampler on the collector?)

to implement this on the instrumentation side, you'd need to propagate the trace start time down to its children, e.g. using baggage

**Reply by srinusanduri** [reply]
@trask Thank you for the response. We have this feature added using baggage in custom agent implementation. But we are now discarding custom agent and going with standard OpenTelemetry agent instead. We are exploring ways of retaining this feature with standard agent. If its not possible to achieve this with standard library, we will explore backend and otel-collector options.
